java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -291,14 +292,33 @@ public static void assertPartialTablesAreEqual(Table expected, long rowOffset, l
       if (rowOffset != 0 || length != expected.getRowCount()) {
         name = name + " PART " + rowOffset + "-" + (rowOffset + length - 1);
       }
-      assertPartialColumnsAreEqual(expect, rowOffset, length, cv, name, enableNullCheck);
+      assertPartialColumnsAreEqual(expect, rowOffset, length, cv, name, enableNullCheck, null);
+    }
+  }
+
+  public static void assertPartialTablesAreEqual(Table expected, long rowOffset, long length, Table table,
+                                                 boolean enableNullCheck, HostColumnVector.ColumnBuilder.DataType type) {
```

Comment:
type does not work here generically.  For this particular API call type is the type of any and all structs that are in the table.  That is not maintainable.  There can be lots of tests with multiple structs in them, or nested structs.  We should either be able to determine the schema of the data from the column itself, or we should have this take in the schema for the entire table, not just a single column.

Reply:
That is true :( Sorry I missed that, Let me fix this!